### PR TITLE
Add clarinet check step to romeo test script

### PR DESCRIPTION
**Pull request by AshtonStephens**
## Summary of Changes

This change adds clarinet check step to romeo test script.

### Risks

Almost no risks. Maybe it flags good clarity contracts as bad.

### How were these changes tested?

Locally running this in repository root works.
```bash
cd ./romeo/asset-contract
./scripts/test.sh
```

### What future testing should occur?

None, this is a testing change.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by friedger** [comment]
My understanding of clarinet is that `clarinet test` is a superset of `clarinet test`.

Does this call detect the syntax error we had?

**Review by CAGS295 — DISMISSED** [review]

**Comment by friedger** [comment]
The syntax was interoduced in https://github.com/stacks-network/sbtc/commit/929250b2622fc25e8dbf0ea001cada4f54cee6b4 and all tests passed: https://github.com/stacks-network/sbtc/actions/runs/6430869284/job/17462753822

File: romeo/asset-contract/scripts/test.sh [hunk 1]
```diff
@@ -2,5 +2,6 @@
 mkdir -p .test
 mkdir -p .coverage
 rm -r .test/*
+clarinet check
```
[the remark this answers is not in the document]
Comment (AshtonStephens) [reply romeo/asset-contract/scripts/test.sh:5]: Right, but the workflow assumes that the working directory is [`romeo/asset-contract`](https://github.com/stacks-network/sbtc/blob/main/.github/workflows/pull-request.yml#L96). Does the rest of the script not also require that?

[the remark this answers is not in the document]
Comment (AshtonStephens) [reply romeo/asset-contract/scripts/test.sh:5]: Just checked - can confirm the whole script is written with that assumption. I see arguments for either running this in root or running it from the directory, I'll defer to you both.

**Comment by AshtonStephens** [comment]
> The syntax was interoduced in [929250b](https://github.com/stacks-network/sbtc/commit/929250b2622fc25e8dbf0ea001cada4f54cee6b4) and all tests passed: https://github.com/stacks-network/sbtc/actions/runs/6430869284/job/17462753822

Exactly, with this check it doesn't pass.

Actually, there is an error here - the script is deemed successful even when it fails. Fixing that..

**Comment by friedger** [comment]
I created an issue for clarinet: https://github.com/hirosystems/clarinet/issues/1191

File: romeo/asset-contract/scripts/test.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (friedger) [reply romeo/asset-contract/scripts/test.sh:5]: best to run it in /romeo/asset-contract

**Review by friedger — APPROVED** [review]
LGTM

Better save than sorry

**Review by CAGS295 — APPROVED** [review]

**Pull request merged**